api/server/router/image/image_routes.go

```diff
@@ -175,7 +175,7 @@ func (ir *imageRouter) postImagesPush(ctx context.Context, w http.ResponseWriter
 		var err error
 		authConfig, err = registry.DecodeAuthConfigBody(r.Body)
 		if err != nil {
-			return errors.Wrap(err, "bad parameters and missing X-Registry-Auth")
```

Comment:
DecodeAuthConfigBody above already returns a typed error; even if wrapped , the API should be able to use that, so changing here should not be needed https://github.com/moby/moby/blob/980d8e5d54d8df31ba0e75e1d9dd474d9d91308a/api/types/registry/authconfig.go#L73-L82